src/app/modules/project/details/details.component.ts

```diff
@@ -220,6 +231,72 @@ export class DetailsComponent implements OnInit {
     });
   }
 
+  /**
+   * Method to display the edit project button based on the current user and the project user.
+   * If the user either has the ProjectWrite scope or is the creator of the project
+   * @param project The project to check if the current user is the owner.
+   */
+  private determineDisplayEditProjectButton(): void {
+    if (this.authService.currentBackendUserHasScope(scopes.ProjectWrite)) {
+      this.displayEditButton = true;
+      return;
+    }
+
+    if (this.currentUser == null || this.project == null || this.project.user == null) {
+      this.displayEditButton = false;
+      return;
+    }
+    this.displayEditButton = this.project.user.id === this.currentUser.id;
+  }
+
+  /**
+   * Method to display the delete project button based on the current user and the project user.
+   * If the user either has the ProjectWrite scope or is the creator of the project
+   * @param project The project to check if the current user is the owner.
+   */
+  private determineDisplayDeleteProjectButton(): void {
+    if (this.authService.currentBackendUserHasScope(scopes.ProjectWrite)) {
+      this.displayDeleteProjectButton = true;
+      return;
+    }
+
+    if (this.currentUser == null || this.project == null || this.project.user == null) {
+      this.displayDeleteProjectButton = false;
+      return;
+    }
+    this.displayDeleteProjectButton = this.project.user.id === this.currentUser.id;
+  }
+
+  /**
+   * Method to display the highlight buttons based on the current user.
+   * If the user either has the HighlightWrite scope.
+   */
+  private determineDisplayHighlightButton(): void {
+    if (this.authService.currentBackendUserHasScope(scopes.HighlightWrite)) {
+      this.displayHighlightButton = true;
+      return;
+    }
+    this.displayHighlightButton = false;
+  }
+
+  /**
+   * Method to display the embed button based on the current user and the project user.
+   * If the user either has the EmbedWrite scope or is the creator of the project
+   * @param project The project to check if the current user is the owner.
```

Comment:
Method has no params, but documentation does?

Reply:
Same goes for line 255

Reply:
That was a leftover comment from development, but ill remove them